cmd/bcpd/app/codec.proto

```diff
@@ -27,13 +41,11 @@ message Tx {
     multisig.UpdateContractMsg update_contract_msg = 9;
     // validators actions
     validators.SetValidatorsMsg set_validators_msg = 10;
+    BatchMsg batch_msg = 11;
```

Comment:
Is this supposed to be recursive? A BatchMsg contains multiple BatchMsg that contain multiple BatchMsg …?

Reply:
I think you are right, we need to implement a manual validation there, so I guess it's best if we separate tx and batchMsg sum stuff